### DISCUSSION TITLE
Project Update

### DISCUSSION THREAD

**Discussion by jawadnasar** [thepercyj/Puzzle-Realms#18] [Q&A]
Hi team, 

Can you tell me about the tasks given to you as the start of the project and during the project as we have changed some roles as well, according to the projects nature. 

Search for all the possible solutions in any data structure format 				Junkai
Use the data strc to store it in DB								                        Jawad
Front-end Design like an web based game rather than a web page				Junkai, Rohan
New Project Plan for this Task									                       Tom, Mya
Timeline Chart for this project									               Jawad, Mya
Search algorithm for getting solutions from the DB and combining Front-end and Back-end		Aman, Tom
Research what is Github Workflows, CI/CD Tools  and how to use it in our Project		                Rohan, Aman
QA												                                                                Rimjhim & Veerpal

**Comment by jawadnasar** [comment]
@JUNKAIOU please update about the design, is it fully complete?

**Comment by Rez27** [comment]
Hi @jawadnasar ,
Till now I have been actively working on the Back-end part with the help of @thepercyj & @tnaccarato. Let me summarize what we did till now since announcement of the project to till date.

- Studied and Implemented Kanoodle Problem using Brute force algorithm (Junkai)
- Researched and understood a better algorithm other than brute force to solve kanoodle problem for given set of custom pieces. (Aman, Rohan)
- Took inspiration from Knuth's Algorithm in Java to solve a different kanoodle problem (Aman, Rohan & Tom)
- Build python code for custom set of pieces and implemented Knuth's algorithm. (Aman, Rohan & Tom)
- Implemented rotations and reflections on the solution set to generate  & ensure complete solutions (Aman, Rohan & Tom)
- Combining all separate functions regarding the puzzle solver into a django project (Aman, Rohan & Tom)


Summarizing, we 3 intensively worked on the backend part of this Puzzle solver right from ensuring completeness & responsiveness.

**Reply by jawadnasar** [reply]
Hi @Rez27, 

During the start of the project we had assigned some tasks to every person. During the project we had some changes in the roles and work idea. I have maintained all the timeline and uploaded it to the repository. If anyones work seems as faded in it, please tell me to give them a full credit in it as well.

**Comment by Mya2328** [comment]
I added some data for new project plan. @jawadnasar Where can I see the timeline Chart?

**Reply by jawadnasar** [reply]
Hi @Mya2328 , 

I have uploaded the timeline in the repository, since we were maintaining it in the excel file. But here in the "repository>timeline" you can see it.

**Reply by Mya2328** [reply]
Hi @jawadnasar ,
I have seen the well organized timeline. It seems like I don't need to add additional information.

**Comment by thepercyj** [comment]
Hi Team,

I have just added local DB connection files and strings, just incase production DB has some problems, you can uncomment the local settings and comment the production DB settings to run the server, configuration files has been updated on my branch as same.

**Comment by thepercyj** [comment]
Hi Team,

Just had some major update to our github repo regarding the polysphere app images which were stored locally that was causing slow response times to push, pull and cloning. Now, the local media files have been sucessfully migrated to our cloud bucket where it will fetch images from the servers. Hence, saving a huge storage space consumed in our repo. The following activities were done.

- Created Object Storage in the Cloud (i.e. Bucket) to store all the image solutions so to reduce 2GB storage consumed in our repo that is creating lot of issues of slow times when pulling, cloning or pushing requests.

- Cleaned some unused variables that were not relevant to the app.

- Commented media directory settings from settings.py as now we don't need to keep local copy of it.

- Commented previous local directory using commands in kanoodle.js and views.py so now it reads from oracle bucket url path as read request have been enabled in the cloud.

**Comment by Mya2328** [comment]
Hi team,
I created the first draft of our reflective essay. As a draft, I still need to add required details and delete some points. I still need to reduce the word counts. Can you please check and give me an advice to write a final reflective essay?

[Reflective Essay for ASE group (6).docx](https://github.com/thepercyj/ASE-Group-6/files/13565060/Reflective.Essay.for.ASE.group.6.docx)

**Reply by thepercyj** [reply]
Hello @Mya2328 ,

The reflective essay looks good however you can crunch it down into relevant point format where it is easier for the user to understand, since people really dont like reading long senteces so what you can do is divide headings into 5 tasks, 1 till 5 just like how you did for Nqueens and Kanoodle do the same for task 3, 4 and 5 and highlights major points of the tasks, rest of the components regarding backend and frontend looks good however you can still make point format over here as well. make sure to make it descrptive and to the point.

Nice work on the documentation. !!! keep up the work.

**Reply by thepercyj** [reply]
Also @Mya2328, you have mentioned the QA has done a lot of error fixes and made a proper guideline to understand how our program is working, so make sure if you mention this point you have a user manual type of documentation either as a doc file or posted in the readme of github for users to understand how to use the project and things of that nature. Remember whatever you claim in the reflective essay, you need to prove it to the professor by giving supporting evidence as well. 

Thanks for the effort once again. !!!!

**Comment by thepercyj** [comment]
Hi @jawadnasar ,

whats the update on the gantt chart, the documentation team has already finished the reflective essay in some form, they might require the picture so they can put it in the essay. Let us know of it.

Thanks !!

**Comment by Mya2328** [comment]
Hi team,
This is the final reflective essay but we need to add QA testing for Task 5. Could you please check my essay and let me know if there is some data missing.
[Reflective Essay for ASE group (6)(Final ).docx](https://github.com/thepercyj/ASE-Group-6/files/13614753/Reflective.Essay.for.ASE.group.6.Final.docx)

**Reply by thepercyj** [reply]
Hi @Mya2328,

Looks great, great work !!!!

**Comment by thepercyj** [comment]
Hi Team,

Ive done some major updates for Task 5:

- Converged all django apps into one big django project titled portal
- added frontend and backend logic for Task 5 
- Added wallpapers, themes, beautify content
- Added user authentication schema as well
- Added page route for all apps to landing.html page 
- Added register and login function 
- Fixed some minor errors in each of the tasks

**Comment by Mya2328** [comment]
Hi Team,
I uploaded the reflective essay in my branch. Can you guys please check the grammar and writing style and the contents in my essay? I would like to make sure to mention all the experience we got throughout our projects. If there is something missing, tell me. Before we submit it, I can rewrite it until all the members are satisfied. 
@jawadnasar 
@thepercyj 
@Rez27 
@veerpalkaur98 
@tnaccarato 
@Togotogo98

**Reply by thepercyj** [reply]
Hi @Mya2328,

Looks great, great work !!!! The team has added some comments on the report make sure to adhere and make necessary changes to it before final submission. Thanks !!!

**Comment by Mya2328** [comment]
Hi Team,
I added the descriptions for Task 4 and 5 in the main page(README.md). Please check and do let me know if there is something to add .I pushed it only to my branch.
@jawadnasar 
@thepercyj 
@Rez27 
@tnaccarato 
@Togotogo98 
@JUNKAIOU
@veerpalkaur98

**Reply by thepercyj** [reply]
Hi @Mya2328,

Looks great, great work !!!!

**Comment by Mya2328** [comment]
Hi Team,
Our final reflective essay is almost done. We only need to add the QA test for the Task 5. You can see the updated reflective essay in my branch. I created draft folder to save all the drafts.
@Rez27 
@thepercyj 
@tnaccarato 
@jawadnasar 
@JUNKAIOU 
@Togotogo98 
@veerpalkaur98

**Reply by thepercyj** [reply]
Hi @Mya2328,

I will share the details accordingly.

**Comment by Mya2328** [comment]
Hi team,
Please check the content writing in our blog. I pushed it to my branch. 

@thepercyj 
@Rez27 
@tnaccarato 
@jawadnasar 
@Togotogo98 
@JUNKAIOU 
@veerpalkaur98

**Reply by thepercyj** [reply]
Hi @Mya2328,

Looks great, great work !!!! 👌👍👍

**Comment by thepercyj** [comment]
Hi Team, 

Great presentation today to the teaching assistant, we were effective at delivering all our objectives on time which makes us ahead of all the other group's schedule. The only thing left is to make some further adjustment in our Task 4 pyramid, where we just need to restrict the pieces from getting colored or make a skew angle, so lets make a final push and lets end this module with a banger. Thanks to everyone who has contributed in this project so far. It was an honour working with you guys.. !!!. Although, we still need to maintain this in the near future as we wanna bring some value to our project.

Thank you all !! Hope we can carry this energy probably in the next sem and achieve our success in whatever we embark on... ❤️
@Rez27 
@tnaccarato 
@jawadnasar 
@Mya2328 
@JUNKAIOU 
@Togotogo98 
@veerpalkaur98

**Comment by thepercyj** [comment]
Hi Team,

Thank you for all of you who have joined in this roller coaster ride, this project would not be completed without all of your hardwork, restless nights, vision and passion for success. I thank you everyone from the bottom of my hard for such a beautiful collaboration. Hope to work with you guys in many projects in the future.

Lastly, thanking our professor @hsimho, for grouping this team and his constant guidance and support during the project. I am honoured to have team members like you. it seems stars did align for us in this project.

Thank you once again !!! ❤️❤️❤️😁
@Rez27 
@tnaccarato 
@jawadnasar 
@Mya2328 
@Togotogo98 
@JUNKAIOU 
@veerpalkaur98
